aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,298 @@
+:version: 1.0.0
+
+Hosting
+=======
+
+By `Steve Smith`_
+
+To run an ASP.NET Core app, you need to configure and launch a host using ``WebHostBuilder``.
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+`View or download sample code <https://github.com/aspnet/Docs/tree/master/aspnet/fundamentals/hosting/sample>`__
+
+What is a Host?
+---------------
+
+ASP.NET Core apps require a *host* in which to execute. A host must implement the :dn:iface:`~Microsoft.AspNetCore.Hosting.IWebHost` interface, which exposes collections of features and services, and a ``Start`` method. The host is typically created using an instance of a :dn:class:`~Microsoft.AspNetCore.Hosting.WebHostBuilder`, which in builds and returns a  :dn:class:`~Microsoft.AspNetCore.Hosting.WebHost` instance. The ``WebHost`` has a private ``Server`` property that is used once the application is up and running to handle requests. Learn more about :doc:`servers <servers>`.
+
+What is the difference between a host and a server?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The host is responsible for ensuring the application's services and the server are available and properly configured. You can think of the host as being a wrapper around the server. The host is configured to use a particular server; the server is unaware of its host.
+
+Setting up a Host
+-----------------
+
+You create a host using an instance of ``WebHostBuilder``. This is typically done in your app's entry point, which by default will be located in a *Program.cs* file. A typical *Program.cs*, shown below, demonstrates how to use a ``WebHostBuilder`` to build a host. 
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/Program.cs
+  :emphasize-lines: 14-21
+  :language: c#
+  :caption: Program.cs
+
+The ``WebHostBuilder`` is responsible for creating the host that will bootstrap the server for the app. ``WebHostBuilder`` supports many optional extensions, but the specification of an ``IServer`` is required. The built-in server is Kestrel. Use ``UseKestrel`` (with options, if desired) to set it up as your app's server.
+
+The server's *content root* determines where it searches for content files, like MVC View files. The default content root is the folder from which the application is run. Specifying ``Directory.GetCurrentDirectory`` as the content root will use the web project's root folder as the app's content root when the app is started from this folder (for example, calling ``dotnet run`` from the web project folder).
```

Comment:
>Specifying ``Directory.GetCurrentDirectory`` as the content root will use the web project's root folder as the app's content root when the app is started from this folder (for example, calling ``dotnet run`` from the web project folder). 

``.UseContentRoot(Directory.GetCurrentDirectory())`` sets the content root to the web apps root  folder.  

Actually I think you should remove this, the code is more readable than the english. You could say

``.UseContentRoot(Directory.GetCurrentDirectory())``  works for most apps. It works for VS/IIS Express, when you call ``dotnet run`` from the web project folder, when you deploy to Azure.  But that's not very useful information. It would be more useful if you told them when it didn't work or where it would be useful to change. I'm guessing leaving it out is the best thing. @danroth27